monai/data/image_reader.py

```diff
@@ -689,8 +694,10 @@ def __init__(self, backend: str = "OpenSlide", level: int = 0):
             self.wsi_reader, *_ = optional_import("openslide", name="OpenSlide")
         elif self.backend == "cucim":
             self.wsi_reader, *_ = optional_import("cucim", name="CuImage")
+        elif self.backend == "tifffile":
```

Comment:
I would suggest to change this `optional_import` to the beginning of this file and add the **dependency check** like:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/data/image_reader.py#L135
Maybe it's not very clear to you, I can do that for `WSIReader` later when this PR merged.

Thanks.

Reply:
Thanks for your comment @Nic-Ma. I checked the implementation for for `@require_pkg` and it seems that it cannot handle the situation here, where the required package depends on the requested `backend`. So it either depends on `cuCIM` or `OpenSlide` or `Tifffile` but not all of them.

Reply:
Hi @drbeh ,

I submitted PR https://github.com/Project-MONAI/MONAI/pull/3312 for this enhancement.

Thanks.